### ISSUE TITLE
Release tarballs misdetect python 3.10

### ISSUE THREAD

**Issue by floppym** [karelzak/util-linux#1319]
Building from a tarball fails with python 3.10 due to a broken automake macro. This has been fixed in the [latest stable release of automake (1.16.3)](https://savannah.gnu.org/forum/forum.php?forum_id=9862).

Please ensure that the latest version of automake is installed on your system when generating future tarballs.

Gentoo bug report: https://bugs.gentoo.org/791196

**Comment by karelzak** [comment]
Hmm... let's blame Fedora 33 ;-)   ... where is still 1.16.2.

Anyway, thanks for your feedback!  I'll update my system ASAP.

**Comment by karelzak** [comment]
It seems Fedora (33 and 34) already contains a patch for this issue, not sure why you see the problem ... I need to test it.

**Comment by floppym** [comment]
As a quick check, take a look at the generated configure script. The problematic line is this:
```
  am_cv_python_version=`$PYTHON -c "import sys; sys.stdout.write(sys.version[:3])"`
```
This causes "3.10" to be cut off as "3.1".

With automake 1.16.3 (or a backported fix), you should see this:
```
  am_cv_python_version=`$PYTHON -c "import sys; print('%u.%u' % sys.version_info[:2])"`
```

**Comment by karelzak** [comment]
Thanks for the details! 

It seems the current Fedora (with automake-1.16.2-4.fc34) produces a proper macro, but the version I had to generate -rc2 had the old problematic line.
```bash
$ ./autogen.sh; ./configure
...
$ grep  'am_cv_python_version=' ./configure
  am_cv_python_version=`$PYTHON -c "import sys; print('%u.%u' % sys.version_info[:2])"`
```

**Comment by karelzak** [comment]
Fixed in the final v2.37.
